software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/AbstractMongoDBSshDriver.java

```diff
@@ -151,24 +186,38 @@ protected String getConfFile() {
         return getRunDir() + "/mongo.conf";
     }
 
+    protected String getRootUsername() {
+        return entity.config().get(AbstractMongoDBServer.ROOT_USERNAME);
+    }
+
     protected ImmutableList.Builder<String> getArgsBuilderWithDefaults(AbstractMongoDBServer server) {
-        Integer port = server.getAttribute(MongoDBServer.PORT);
+        ImmutableList.Builder<String> builder = getArgsBuilderWithNoAuthentication(server);
+        if (MongoDBAuthenticationUtils.usesAuthentication(entity)) {
+            builder.add("--keyFile", entity.getAttribute(AbstractMongoDBServer.MONGODB_KEYFILE_DESTINATION));
+        }
+        return builder;
+    }
 
-        return ImmutableList.<String>builder()
-                .add("--config", getConfFile())
-                .add("--pidfilepath", getPidFile())
-                .add("--logpath", getLogFile())
-                .add("--port", port.toString())
-                .add("--fork");
+    protected ImmutableList.Builder<String> getArgsBuilderWithNoAuthentication(AbstractMongoDBServer server) {
+        Integer port = server.getAttribute(MongoDBServer.PORT);
+        ImmutableList.Builder<String> builder = ImmutableList.builder();
+        builder.add("--config", getConfFile());
+        builder.add("--pidfilepath", getPidFile());
+        builder.add("--logpath", getLogFile());
+        builder.add("--port", port.toString());
+        builder.add("--fork");
+        return builder;
     }
     
     protected void launch(ImmutableList.Builder<String> argsBuilder) {
         String args = Joiner.on(" ").join(argsBuilder.build());
-        String command = String.format("%s/bin/mongod %s > out.log 2> err.log < /dev/null", getExpandedInstallDir(), args);
-        LOG.info(command);
+        List<String> commands = new LinkedList<String>();
+        commands.add(String.format("%s/bin/mongod %s > out.log 2> err.log < /dev/null", getExpandedInstallDir(), args));
```

Comment:
It's worth changing the stdout/err redirections to append to the files rather than overwrite them so we can keep the logs from the create-user script.